### ISSUE TITLE
Feature Request: Allow csv search() to match on multiple values

### ISSUE THREAD

**Issue by nap2000** [getodk/collect#1757]
#### Software and hardware versions 
Collect v1.12.2, Android vx.x.x, device used...

#### Problem description
My understanding of the current implementation of search() is that you can only match a single (optionally wildcarded) value when searching a CSV file for choices. Plus it allows a second filter.

However it would be useful to allow multiple distinct values to be matched.  For example if you select multiple options from a product category list then you may want to see all products in those categories in a second select multiple.  

Alternatively you may want to use a calculate to identify multiple potential codes and then show a select question to allow the user to select choices based on those codes.  

#### Proposed Solution

The code already nearly implements the ability to use multiple values using the following syntax:
`search('csvfile', 'matches',  'col1 col2 col3', 'value1 value2 value 3')`
After a minor change to get this to work the above results in all rows where col1=value1 or col2=value2 or col3=value3.

I propose also to add a new matching function "in" which would match in a single column but for any number of values.
`search('csvfile', 'in', 'col', 'value1 value2 value3')`

The above is more flexible and more useful that the former as values can be dynamically generated from other select questions or calculates.

#### Other information 
I have tested both of the above changes in a fork and provide a pull request if desired.

**Comment by lognaturel** [comment]
I don't use `search()` myself instead relying on external itemsets or external secondary instances. `search()` exists pretty far out of the XForms spec so it's not usable in clients across the ecosystem, etc. All that to say that I'm not really the best person to evaluate this change! It would be great to get other folks who rely on the appearance/function to chime in.

My initial reaction is that I'm nervous about the convention that values are separated on spaces.

`search('csvfile', 'matches', 'col1 col2 col3', 'value1 value2 value 3')`
I can currently use spaces in the match string as in `search('csvfile', 'matches', 'comments', 'No comments')`, right? Would the rule with what you're proposing then be that if there are spaces in the column name string the match string should be split on spaces? So `search('csvfile', 'matches', 'comments user', 'No comments')` Would search for a comment of `No` and a user called `comments`?

It also seems a odd to me that the conditions are implicitly joined with `or` and that there is no `and` equivalent.

If the above were added, `search('csvfile', 'in', 'col', 'value1 value2 value3')` could be written as `search('csvfile', 'matches', 'col col col', 'value1 value2 value 3')`, right?

**Comment by nap2000** [comment]
Yes I agree with your comments on spaces and implicit joining of columns with 'or'.   Hence I rowed back from that and in my final implementation only added the "in" function.  This allowed for the construction:

    select_multiple category    search('products')
    select_multiple product     search('products', 'in', 'category', ${category})

This returns all the products in all the categories selected by the user. Which is all I needed for a customer implementation.

It may be possible to make something of this multi-column search but it may also be best to just let it lie dormant.  

Yes "search('csvfile', 'matches', 'col col col', 'value1 value2 value 3')" would be the equivalent of "in" however it would be tedious to construct to meet the example above.  And as discussed above the multi column approach would need some more serious thought.

**Comment by nap2000** [comment]
Just to confirm the implemented "in" method assumes all values are space separated.  Great if they have come from a select multiple in your form, a constraint that you will need to live with if they have come from a csv file, a text question or a calculate etc.  Since "in" is new that constraint did not worry me too much.

**Comment by lognaturel** [comment]
Ah, how interesting! I'm a lot more sympathetic to the idea seeing this concrete example using the result of a previous `select_multiple`, thanks for sharing that. Will keep pondering it and hope that others chime in. 

It would be great to have a more general-purpose approach to this complex querying. That's what external secondary instances do by allowing for full XPath querying but they're not performant enough for large datasets yet (though @dcbriccetti is working on it).
